BUGFIX: Make session caches persistent by default

albe opened the pull request:
Previously the sessions would be flushed whenever caches get flushed. This sometimes leads to unexpectedly logging out users, as happens for example in the neos setup in the last step.
This is only breaking in the sense, that you need to take care when deploying code that changes classes or data stored in the session. In those cases you need to flush the session caches manually via the new CLI command `./flow flow:session:destroyAll`. Also see the documentation https://flowframework.readthedocs.io/en/stable/TheDefinitiveGuide/PartIII/SessionHandling.html#session-storage

Resolves #1390

albe commented:
For the record, I don't target this on 4.3 due to the "breakyness". It's still not breaky enough that I would hold this back until the next major. Up to decide for the RM @jonnitto though :)

kitsunet commented:
Not sure if this kind of change shouldn't go into a major, just for the quite big behavior change.

beheist commented:
I'd be happy with merging it into a minor. To me, the benefits are big enough that I'm fine with merging this in a minor - that setup bug is really extremely confusing to many users, and I'd love to see that fixed.

albe commented:
One thing I noticed, but which I can not fully assess: The persistent cache is not Context-specific, i.e. sessions would be shared among all existing contexts that way. This can be considered a good thing as well as a bad thing, depending on the use case. Since there is a workaround, namely configuring the `cacheDirectory` for the session in each context, I'd say it's okay as long as it is properly documented.

albe commented:
Added a small CLI command controller for destroying sessions more conveniently and added documentation. Also updated the `flow:cache:flush` command documentation to state that sessions are NOT deleted any more, so the Reference section of the docs will get updated when docs are generated from code with the release.

@jonnitto Are you okay with the documentation?
